src/sumo/wrapper/_auth_provider.py

```diff
@@ -0,0 +1,239 @@
+import msal
+import os
+import stat
+import sys
+import json
+import jwt
+import time
+from azure.identity import ManagedIdentityCredential
+
+from msal_extensions.persistence import FilePersistence
+from msal_extensions.token_cache import PersistedTokenCache
+
+if not sys.platform.startswith("linux"):
+    from msal_extensions import build_encrypted_persistence
+
+
+def scope_for_resource(resource_id):
+    return f"{resource_id}/.default offline_access"
+
+
+class AuthProvider:
+    def __init__(self, resource_id):
+        self._scope = scope_for_resource(resource_id)
+        self._app = None
+        return
+
+    def get_token(self):
+        accounts = self._app.get_accounts()
+        result = self._app.acquire_token_silent([self._scope], accounts[0])
+        if "error" in result:
+            raise ValueError(
+                "Failed to silently acquire token. Err: %s"
+                % json.dumps(result, indent=4)
+            )
+        # ELSE
+        return result["access_token"]
+
+    pass
+
+
+class AuthProviderAccessToken(AuthProvider):
+    def __init__(self, access_token):
+        self._access_token = access_token
+        payload = jwt.decode(access_token, options={"verify_signature": False})
```

Comment:
Should we consider re-using the signature verification code we have applied elsewhere for this?

Reply:
I like the pass statements, but that's probably mainly because I think indentation-based program structure is an idiotic idea... the pass statements make the end of a block explicit. 

No need for signature verification; the access token will be verified by sumo-core.